security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/access-termination-notification/files/src/com/oracle/sample/AccessTerminationNotification.java

```diff
@@ -0,0 +1,397 @@
+ * SOFTWARE.
+ *******************************************************************************/
+
+package com.oracle.sample;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.thortech.xl.dataaccess.tcDataProvider;
+import com.thortech.xl.dataobj.util.XLDatabase;
+import com.thortech.xl.dataobj.util.tcEmailNotificationUtil;
+
+import oracle.iam.identity.usermgmt.api.UserManager;
+import oracle.iam.identity.usermgmt.api.UserManagerConstants;
+import oracle.iam.identity.usermgmt.vo.User;
+import oracle.iam.platform.Platform;
+import oracle.iam.platform.entitymgr.vo.SearchCriteria;
+import oracle.iam.provisioning.api.ProvisioningConstants;
+import oracle.iam.provisioning.api.ProvisioningService;
+import oracle.iam.provisioning.vo.Account;
+import oracle.iam.provisioning.vo.EntitlementInstance;
+import oracle.iam.scheduler.exception.SchedulerException;
+import oracle.iam.scheduler.vo.TaskSupport;
+
+/**
+ * 
+ * Sample scheduled task implementation that sends notification emails to users
+ * and their managers prior to user account/entitlement expiration or access
+ * termination.
+ * 
+ * It uses the standard Thortech/Oracle Identity Manager email APIs.
+ * 
+ * @author mgere-oracle
+ * 
+ */
+public class AccessTerminationNotification extends TaskSupport {
+
+	private static final Logger logger = Logger.getLogger(AccessTerminationNotification.class.getName());
+
+	private static final String TEMPLATE_SELECT_QUERY = "SELECT NOTIFICATIONTEMPLATE.ID, LONGMESSAGE, SUBJECT "
+			+ "FROM LOCALTEMPLATE, NOTIFICATIONTEMPLATE WHERE LOCALTEMPLATE.TEMPLATEID = NOTIFICATIONTEMPLATE.ID AND TEMPLATENAME = ?";
+
+	private static final String FROM_EMAIL_ADDRESS = "donotreply@example.com";
+
+	private Date currentDate;
+	private Integer daysBeforeExpiration;
+
+	private StringBuilder expiringItemsText;
+
+	@Override
+	public void execute(HashMap arg0) throws Exception {
+		logger.log(Level.INFO, "The scheduled task run has started.");
+
+		if (arg0 == null) {
+			// arg0 will never be null if a valid scheduled task definition file is used
+			throw new SchedulerException("Scheduled task definition is incorrect. Aborting run.");
+		}
+
+		// retrieve parameters from scheduled task
+		daysBeforeExpiration = Integer.parseInt((String) arg0.get("Days Before Expiration"));
+		String emailTemplateName = (String) arg0.get("Email Template Name");
+
+		UserManager userManager = Platform.getService(UserManager.class);
+		ProvisioningService provisioningService = Platform.getService(ProvisioningService.class);
+
+		currentDate = Calendar.getInstance().getTime();
+		logger.log(Level.FINE, "Current date: {0}", currentDate);
```

Comment:
Use the respective convenience method, i.e., logger.log(Level.FINE) will become logger.fine().
This applies for all logger.log() calls.

Reply:
@oheimburger For the sake of consistency with the codebase we've built so far I would rather not. Considering WebLogic handles logging levels specifically in the Level.LEVEL format, it's a lot easier and clearer to manage logging this way. There's no functional difference or benefit in using the convenience methods.

Reply:
Not a good reasoning. That's why we have so many log4j issues.